system-parachains/people-paseo/src/xcm_config.rs

```diff
@@ -195,13 +268,18 @@ pub type Barrier = TrailingSetTopicAsId<
 					// allow it.
 					AllowTopLevelPaidExecutionFrom<Everything>,
 					// Parent and its pluralities (i.e. governance bodies) get free execution.
-					AllowExplicitUnpaidExecutionFrom<(
-						ParentOrParentsPlurality,
-						FellowsPlurality,
-						Equals<RelayTreasuryLocation>,
-						Equals<AssetHubLocation>,
-						AssetHubPlurality,
-					)>,
+					AllowExplicitUnpaidExecutionFrom<
+						(
+							ParentOrParentsPlurality,
+							FellowsPlurality,
+							Equals<RelayTreasuryLocation>,
+							Equals<AssetHubLocation>,
+							// Because of InitiateTransfer {preserve_origin: true}
+							Equals<RootLocation>,
+						),
+						// Because of InitiateTransfer {preserve_origin: true}
```

Comment:
I don't understand this bit. Why are we adding the root location to this list because of preserving origin?

Reply:
indeed, I completely removed the changes for `Barrier`, I don't know the reason those change were introduced. https://github.com/paseo-network/runtimes/pull/301/commits/a8b9750c433cec8cbd0f2ef38c825b40b8dc4a92